core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -404,12 +397,7 @@ private void removeIndex(String index, String reason, boolean delete) {
             if (delete) {
                 indicesLifecycle.beforeIndexDeleted(indexService);
             }
-            IOUtils.close(Iterables.transform(pluginsService.indexServices(), new Function<Class<? extends Closeable>, Closeable>() {
-                @Override
-                public Closeable apply(Class<? extends Closeable> input) {
-                    return indexInjector.getInstance(input);
-                }
-            }));
+            IOUtils.close(pluginsService.indexServices().stream().map((p) -> indexInjector.getInstance(p)).collect(Collectors.toList()));
```

Comment:
Similarly here, what do you think of the following?

    Stream<Closeable> closeables = pluginsService.indexServices().stream().map((p) -> indexInjector.getInstance(p));
    IOUtils.close(closeables::iterator);

This saves an allocation of an `ArrayList`, and possibly multiple backing arrays as the elements are added to the `ArrayList`.